### PR TITLE
Fix FOUC on Firefox

### PR DESCRIPTION
**Pull request by sondr3**
As mentioned in #1698 Firefox users experience a brief FOUC (flash of unstyled content) while loading various pages in Gitea, this commit should fix this by using Filament Groups loadCSS library to asynchronously load the CSS that makes the pages flash.

This commit will add ~1.5kB of JS to the `<head>` of the webpages. The scripts need to be in the head for the FOUC to not occur, if you do it with the scripts in the bottom of the page it will load the CSS too late and the FOUC will occur regardless. You could potentially also inline them but I went with linking to them.

You will notice in the before and after from the developer panel that the Font Awesome CSS is loaded last because it will not load until the DOM content is ready.
screen shot 2017-05-14 at 19 56 08
screen shot 2017-05-14 at 19 56 23

*NOTE*: In case users don't have JS enabled it will load the CSS as normal. However I'm not sure if this is required and that I could just remove the `<noscript>` line.

This is my first pull request to Gitea, so let me know if I need to change anything. 😃

### CODE REVIEW

**Comment by MCF** [comment]
I've tested this on MacOS as well.  Looks good in Firefox and Chrome.  Did not see any small flashes in Firefox at all, which occasionally show up without this patch applied.

I'll try to find some time to test on Windows 7 in the next few days.

**Comment by MCF** [comment]
Built and tested on Windows 7.  I tested Firefox 53, Chrome 58 and IE 11.  No FOUC on any of them, not even on initial (uncached) page load.  Better than github in that regard.

I did not notice any obvious problems with page load speeds on any of the browsers.  But I just browsed around the site and compared it to try.gitea.io.  No real timing tests.

Nice job with the fix.

File: templates/base/head.tmpl [hunk 1]
```diff
@@ -20,8 +20,9 @@
 {{end}}
 
 	<link rel="shortcut icon" href="{{AppSubUrl}}/img/favicon.png" />
-	<link rel="stylesheet" href="{{AppSubUrl}}/assets/font-awesome-4.6.3/css/font-awesome.min.css">
-	<link rel="stylesheet" href="{{AppSubUrl}}/assets/octicons-4.3.0/octicons.min.css">
+	<link rel="preload" href="{{AppSubUrl}}/assets/font-awesome-4.6.3/css/font-awesome.min.css" as="style" onload="this.rel='stylesheet'">
+  <noscript><link rel="stylesheet" href="{{AppSubUrl}}/assets/font-awesome-4.6.3/css/font-awesome.min.css"></noscript>
+  <link rel="stylesheet" href="{{AppSubUrl}}/assets/octicons-4.3.0/octicons.min.css">
```
Comment (ethantkoenig) [inline templates/base/head.tmpl]: lines 24 and 25 are indented with spaces

**Review by ethantkoenig — CHANGES_REQUESTED** [review]

File: templates/base/head.tmpl (hunk 1 above)
Comment (sondr3) [reply templates/base/head.tmpl]: Oh, you use tabs, I didn't notice, I'll fix it.

Comment (sondr3) [reply templates/base/head.tmpl]: And done 😄

**Comment by lunny** [comment]
So this will fix #1271?

**Comment by MCF** [comment]
Not sure about that one, but it does fix https://github.com/go-gitea/gitea/issues/1698

**Comment by sondr3** [comment]
@lunny, this specifically fixes #1698 and if it fixes anything else that's purely coincidental 😃

**Comment by lunny** [comment]
@sondr3 I have tested that this will fix #1271, so LGTM

**Comment by strk** [comment]
LGTM

**Comment by strk** [comment]
Actually @sondr3  please add the license of the new .js files to https://github.com/go-gitea/gitea/blob/master/public/assets/librejs/librejs.html

**Comment by strk** [comment]
@sondr3 see #1484 (there are also more librejs things missing, in case you want to deal with this in a separate PR)

**Comment by lunny** [comment]
@ethantkoenig please confirm your codereview

**Comment by sondr3** [comment]
@strk I'll add them to this PR when I get to my computer in a few hours and push a new change 😃

**Comment by sondr3** [comment]
Alright, added the JS files to the `librejs.html` file, I wasn't sure where to link to the license since there's not a canonical homepage for it but OSI seemed like a good fit.

**Review by ethantkoenig — APPROVED** [review]

**Pull request merged**